crates/router/src/core/payment_methods/cards.rs

```diff
@@ -3649,6 +3653,14 @@ pub async fn list_payment_methods(
             business_profile.collect_billing_details_from_wallet_connector
         }));
 
+    let is_tax_connector_enabled = business_profile.as_ref().map_or(false, |business_profile| {
+        let is_tax_connector_enabled = business_profile.is_tax_connector_enabled;
+        match &business_profile.tax_connector_id {
+            Some(id) if !id.is_empty() => is_tax_connector_enabled,
+            _ => false,
+        }
+    });
```

Comment:
Can we add this as a method on the business profile domain model instead?

Reply:
Done